packages/nextjs/components/Header.tsx

```diff
@@ -77,6 +78,25 @@ export const Header = () => {
   const { targetNetwork } = useTargetNetwork();
   const isLocalNetwork = targetNetwork.id === devnet.id;
 
+  const { provider } = useProvider();
+  const { address, status } = useAccount();
+  const [isDeployed, setIsDeployed] = useState(false);
+
+  useEffect(() => {
+    if (status === "connected" && address) {
+      provider
+        .getContractVersion(address)
+        .then((v) => {
+          if (v) setIsDeployed(true);
+        })
+        .catch((e) => {
+          console.log(e);
+          setIsDeployed(false);
+        });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [status, address]);
+
```

Comment:
```suggestion
  const { provider } = useProvider();
  const { address, status } = useAccount();
  const [isDeployed, setIsDeployed] = useState(false);

  useEffect(() => {
    if (status === "connected" && address && provider) {
      provider
        .getContractVersion(address)
        .then((v) => {
          if (v) setIsDeployed(true);
        })
        .catch((e) => {
          console.log(e);
          setIsDeployed(false);
        });
    }

  }, [status, address, provider]);

```

Reply:
I have checked that provider doesn't cause unnecessary re rendering and all is ok